### DISCUSSION TITLE
Feature ".tube" , how do i use correctly

### DISCUSSION THREAD

**Discussion by FCalderon3006** [pyvista/pyvista#2417] [Q&A]
**I want to replicate https://github.com/OpenGeoVis/omfvista"

 I extract the points from points=assay.points() and extract the density of the same with df = assay.set_active_scalars('DENSITY') and
 df = assay ['DENSITY'] but i can´t replicate.
I think that i have to do with a dataframe that has  X/Y/Z/from/to/lithology, in the same way that we can create a drillhole data in software.
Thank you in advance

-----

**To Reproduce**

Please include a code snippet to reproduce the bug in the block below:

```py
# Insert code here

```

**Screenshots**

-----

**System Information:**
 Please run the following code wherever you are experiencing the bug and paste the output below. This report helps us track down bugs and it is critical to addressing your bug:

```py
# Get system info
import pyvista as pv
print(pv.Report())
```

```txt
# Paste system info here

```

**Comment by akaszynski** [comment]
Hello,

I'll need a bit more of a code example to be able to help you.

The [tube](https://docs.pyvista.org/api/core/_autosummary/pyvista.PolyData.tube.html) method requires line geometry for it to work. See the [lines](https://docs.pyvista.org/api/core/_autosummary/pyvista.PolyData.lines.html) attribute for PolyData. Note that the construction of lines uses the standard vtk cell approach.
